libpod/options.go

```diff
@@ -613,16 +605,26 @@ func WithSystemd() CtrCreateOption {
 	}
 }
 
+// WithSdNotifySocket sets the sd-notify of the container
+func WithSdNotifySocket(socketPath string) CtrCreateOption {
```

Comment:
We should only allow this is SdNotifyMode is set appropriately; you can validate in `ctr_validate.go`

Reply:
It's done during specgen. Do you still want it in ctr_validate.go?

Reply:
Yes, Libpod should validate independently when possible.